app.py

```
@@ -15,23 +15,26 @@
            methods=['POST'],
            content_types=['application/octet-stream'])
 def exportBag():
-    """Expects a compressed (zipped) BagIt structure that
-    contains two TSV files named  "participant.tsv" and
-    "sample.tsv" in its "data" sub-directory. It creates
-    a new workspace in a FireCloud namespace, and uploads
-    the data into it.
-    Aborts execution and returns HTTP body back to client if any 
-    request operation fails."""
+    """Expects a compressed (zipped) BagIt structure that contains 
+    two TSV files in its "data" sub-directory whose extension needs
+    to be ".tsv". It creates a new workspace in a FireCloud namespace,
+    and uploads the content of the TSV files into it.
+    Aborts execution and returns HTTP body back to client if FC criteria
+    are violated or a request operation fails."""
     req_body = app.current_request.raw_body
     req_query_params = app.current_request.query_params
     req_headers = app.current_request.headers
-    os.chdir('/tmp')
+    path = '/tmp'
+    delete_files_from_path(path)
+    os.chdir(path)
     with zipfile.ZipFile(io.BytesIO(req_body), 'r') as archive:
         archive.extractall()
-    data = {
-        'participant': '/tmp/manifest/data/participant.tsv',
-        'sample': '/tmp/manifest/data/sample.tsv'
-    }
+    # zipfile_name = get_bag_name(os.getcwd())
+    # if isinstance(zipfile_name, Response):
+    #     return zipfile_name  # number of files in zipfile not exactly 1
+    data = get_data(path)  # zipfile_name
+    if isinstance(data, Response):
```

It feels wrong to me to have a method return two totally different types of data; in your case either a Response object or a JSON object.

I would instead throw an Exception from get_data, and catch it. Maybe ask a more experienced Python programmer what they think?